### PR TITLE
CLOUDSTACK-9756: Configure to  ignore the ipassoc failure

### PR DESCRIPTION
**Pull request by jayapalu**
On failure of ip assoc in back end (In VR), added config option to ignore or not to ignore failure. Using this failures won't be ignored and ip address will not set to free in DB.

### CODE REVIEW

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 360
 Hypervisor xenserver
 NetworkType Advanced
 Passed=104
 Failed=1
 Skipped=7

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_non_contigiousvlan.py

 * test_extendPhysicalNetworkVlan Failed


**Skipped tests:**
test_01_test_vm_volume_snapshot
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_routers_network_ops.py
test_disk_offerings.py

**Comment by jayapalu** [comment]
Travis CI test_outofbandmanagement  is failing which is not related to this PR changes.  It seems there is problem with the test case.
@rhtyd  Do you have any idea on this ?
Running tests:  smoke/test_outofbandmanagement smoke/test_over_provisioning smoke/test_password_server smoke/test_portable_publicip smoke/test_primary_storage smoke/test_privategw_acl smoke/test_public_ip_range smoke/test_pvlan smoke/test_regions smoke/test_reset_vm_on_reboot smoke/test_resource_detail smoke/test_router_dhcphosts smoke/test_router_dns smoke/test_routers smoke/test_routers_iptables_default_policy smoke/test_routers_network_ops smoke/test_scale_vm smoke/test_secondary_storage smoke/test_service_offerings smoke/test_snapshots smoke/test_ssvm smoke/test_staticroles smoke/test_templates smoke/test_usage_events smoke/test_vm_life_cycle smoke/test_vm_snapshots smoke/test_volumes smoke/test_vpc_redundant smoke/test_vpc_router_nics smoke/test_vpc_vpn smoke/misc/test_deploy_vm smoke/misc/test_escalations_templates smoke/misc/test_vm_ha smoke/misc/test_vm_sync

Traceback (most recent call last):

  File "/opt/python/2.7.12/lib/python2.7/SocketServer.py", line 596, in process_request_thread

    self.finish_request(request, client_address)

  File "/opt/python/2.7.12/lib/python2.7/SocketServer.py", line 331, in finish_request

    self.RequestHandlerClass(request, client_address, self)

  File "/opt/python/2.7.12/lib/python2.7/SocketServer.py", line 652, in __init__

    self.handle()

  File "/home/travis/.local/lib/python2.7/site-packages/ipmisim/ipmisim.py", line 436, in handle

    return IpmiServerContext().handle(data, address, socket)

  File "/home/travis/.local/lib/python2.7/site-packages/ipmisim/ipmisim.py", line 112, in handle

    self.initiate_session(data, address, self.session)

  File "/home/travis/.local/lib/python2.7/site-packages/ipmisim/ipmisim.py", line 143, in initiate_session

    myaddr, netfnlun = struct.unpack('2B', data[14:16])

error: unpack requires a string argument of length 2

Traceback (most recent call last):

  File "/opt/python/2.7.12/lib/python2.7/SocketServer.py", line 596, in process_request_thread

    self.finish_request(request, client_address)

  File "/opt/python/2.7.12/lib/python2.7/SocketServer.py", line 331, in finish_request

    self.RequestHandlerClass(request, client_address, self)

  File "/opt/python/2.7.12/lib/python2.7/SocketServer.py", line 652, in __init__

    self.handle()

  File "/home/travis/.local/lib/python2.7/site-packages/ipmisim/ipmisim.py", line 436, in handle

    return IpmiServerContext().handle(data, address, socket)

  File "/home/travis/.local/lib/python2.7/site-packages/ipmisim/ipmisim.py", line 112, in handle

    self.initiate_session(data, address, self.session)

  File "/home/travis/.local/lib/python2.7/site-packages/ipmisim/ipmisim.py", line 143, in initiate_session

    myaddr, netfnlun = struct.unpack('2B', data[14:16])

error: unpack requires a string argument of length 2

**Comment by rhtyd** [comment]
@jayapalu thanks, yes it seems an intermittent issue with `ipmisim`, `pyghmi` library that is used by the test to simulate a live ipmi server/board. I'll try to fix it soon, meanwhile you can push -f or close/open the PR to re-kick Travis.
Exception: Job failed: {jobprocstatus : 0, created : u'2017-02-09T13:17:54+0000', cmd : u'org.apache.cloudstack.api.command.admin.outofbandmanagement.ChangeOutOfBandManagementPasswordCmd', userid : u'9f69561f-eec8-11e6-8c34-42010a0a0806', jobstatus : 2, jobid : u'752cba20-e8f8-4b53-9675-8aa98186ee01', jobresultcode : 530, jobresulttype : u'object', jobresult : {errorcode : 530, errortext : u'Failed to change out-of-band management password for host (40328dc0-d290-481f-a6cc-f084ecf88661) due to driver error: Failed to find IPMI user to change password, error: > Error: no response from RAKP 1 message\n> Error: no response from RAKP 1 message\n> Error: no response from RAKP 1 message\n> Error: no response from RAKP 1 message\nSet Session Privilege Level to ADMINISTRATOR failed\nError: Unable to establish IPMI v2 / RMCP+ session\n'}, accountid : u'9f693066-eec8-11e6-8c34-42010a0a0806'}

File: engine/components-api/src/com/cloud/network/IpAddressManager.java [hunk 1]
```diff
@@ -43,6 +43,9 @@
         "If true, when account has dedicated public ip range(s), once the ips dedicated to the account have been consumed ips will be acquired from the system pool",
         true, ConfigKey.Scope.Account);
 
+    static final ConfigKey<Boolean> RulesContinueOnError = new ConfigKey<Boolean>("Advanced", Boolean.class, "network.rule.delete.ignoreerror", "true",
```
Comment (srinivas-gandikota) [inline engine/components-api/src/com/cloud/network/IpAddressManager.java]: The default for ignoreerror is set to true, means the errors will be ignored.
Shouldn't the default be set to false ?

Comment (jayapalu) [inline engine/components-api/src/com/cloud/network/IpAddressManager.java]: To continue with the existing behavior it is set to true. If some one want to enable they can set to false.

Comment (srinivas-gandikota) [inline engine/components-api/src/com/cloud/network/IpAddressManager.java]: sound appropriate.

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 417
 Hypervisor xenserver
 NetworkType Advanced
 Passed=105
 Failed=0
 Skipped=7

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**

**Skipped tests:**
test_01_test_vm_volume_snapshot
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_routers_network_ops.py
test_disk_offerings.py

**Comment by kishankavala** [comment]
Code LGTM

**Comment by jayapalu** [comment]
There are two LGTMs and there are not test failures in the test results. So marking tag:mergeready

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Pull request merged**